static/js/compose.js

```diff
@@ -423,6 +423,12 @@ export function initialize() {
     });
 
     $("#compose-textarea").on("input propertychange", () => {
+        if ($("#compose-textarea").hasClass("invalid")) {
```

Comment:
I added this if statement to avoid fetching the message content for each character the user types.

Reply:
I think we already do that via the `compose_text_length` value returned by `compose_validate.check_overflow_text`; so I suspect if you reorder this check, you'll find that we can reuse that calculation rather than doing this?

(I think it already trims whitespace-only messages when doing that calculation).

Reply:
Great point! I confirmed it trims whitespace as part of the calculation by logging its value, and made that change.